### ISSUE TITLE
Can not parse `create trigger` (postgres) with a fully qualified function name

### ISSUE THREAD

**Issue by NBprojekt** [sqlfluff/sqlfluff#3348]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting postgres trigger does not works as expected. 
```
Found unparsable section: 'create trigger table_before_change befor...'
```

Even the `postgres_create_trigger.sql` can not be linted without issues.

### Expected Behaviour

Should parse create trigger for postgres.

### Observed Behaviour

> Lets ignore `Line is too long`

Starting lining sql, using sqlfluff@0.13.1, target: ./src/test.sql
== [src\test.sql] FAIL
 L: 70 | P: 1 | PRS | Line 70, Position 1: Found unparsable section: 'create
 | trigger table_before_change befor...'
 L: 75 | P: 131 | L016 | Line is too long.
 L: 76 | P: 82 | L016 | Line is too long.
 L: 77 | P: 1 | PRS | Line 77, Position 1: Found unparsable section: 'create
 | or replacte constraint trigger fo...'
 L: 78 | P: 128 | L016 | Line is too long.
 All Finished!

### How to reproduce

[online sqlfluff](https://online.sqlfluff.com/fluffed?sql=H4sIAGhag2IC_62TwW7CMAyG70i8g48grXDftGep3MRpI6VJ57gq7Onn0FZswGFCXCLbcr78-Z1UFfRnMEwoBMK-bYm3m785CDaB6oZcYqpNh7ElmDPwMRMLJIZxsGVPipBNRz0eLrtAu4DQdMBpAjqRGbXJjdGIv2ldyHa3_9huqgryVwAXRufeQSjL0fmTjEz5aD0GMpKPQ8rSlsoa1LPwehF-UMTdZVxKgE40WrVHaJDvtQ2czA65zUXPA8iNA09SZimrea5A3nT5fhqoeoTQFpalQPNICslx6i9kS46YL9Px0Yu6Gc5Lkex1YFkU3VOUF1kzdRRhh_AJzf5_SFXCNAQ0Gpuk92L0Kuf1Q3hI4eIIReNjCynY-RcA5pLUcxJpupY1Wcrr-eVYsvpmfwv4AZKnQ1ZyAwAA&dialect=postgres)

> test.sql
-- my create trigger
create trigger table_before_change before insert or update on schema.table for each row execute function schema.table_changed();
-- sql fluff: test/fixtures/dialects/postgres/postgres_create_trigger.sql
create trigger foo after insert on bar execute function proc(args);
create trigger foo before insert on bar execute function proc(args);
create trigger foo after update of bar, baz on bar execute function proc(args);
create trigger foo instead of delete on bar from baz deferrable initially deferred for each statement execute function proc(args);
create trigger foo before insert on bar when (a = b) execute function proc(args);
create or replacte constraint trigger foo before insert on bar execute function proc(args);
create trigger foo before insert on bar referencing old table as old_table new table as new_table execute procedure proc(args);

### Dialect

postgres

### Version

0.13.1

### Configuration

[sqlfluff]
dialect = postgres
sql_file_exts = .sql

[sqlfluff:rules]
tab_space_size = 2
max_line_length = 80
indent_unit = space
comma_style = trailing

[sqlfluff:rules:L007]
operator_new_lines = before

[sqlfluff:rules:L057]
unquoted_identifiers_policy = all
quoted_identifiers_policy = all
allow_space_in_identifier = true
additional_allowed_characters = "-."

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
> Even the postgres_create_trigger.sql can not be linted without issues.

The intent of fixtures is not to pass all linting tests. Basically we'd rather have the test cases, than make people write test cases that pass all rules (especially as many rules may not be used by certain teams,  for certain syntax).

However, they should:
- Parse for that dialect. I've confirmed this does parse.
- Linting rules should run - i.e. not crash out. Again I've confirmed that is the case.

There's are two issues with your test:
1. We currently do not support fully qualified function names in triggers (e.g. we support `execute function table_changed();` but not `execute function schema.table_changed();`).
2. You have a typo in `create or replacte constraint trigger foo` (`replacte` not `replace`).

To fix the first we should replace these lines:

https://github.com/sqlfluff/sqlfluff/blob/dcfbf55634df3e4ef64d4b4b139ed2dde2a6ee16/src/sqlfluff/dialects/dialect_postgres.py#L3245-L3250

With this:

```py
        Sequence(
            "EXECUTE",
            OneOf("FUNCTION", "PROCEDURE"),
            Ref("FunctionSegment"),
        ),
```

And then add a test case to `test/fixtures/dialects/postgres/postgres_create_trigger.sql`. This is also the way the `CreateMaterializedViewStatementSegment` segment is set up.

Simple enough change.

**Comment by tunetheweb** [comment]
Fix proposed in #3349 . Aiming for a release on Friday so will include this there.
